lib/storage/data/external/GcpClient.js

```diff
@@ -13,7 +13,8 @@ const { ListObjectsCommand,
     GetObjectCommand,
     DeleteObjectCommand,
     CopyObjectCommand,
-    HeadObjectCommand } = require('@aws-sdk/client-s3');
+    HeadObjectCommand,
+    AccessDenied } = require('@aws-sdk/client-s3');
```

Comment:
Errors are not exposed by the lib, from what I can see `AccessDenied` would here always be undefined
It should be `AccessDeniedException` (cf https://aws.amazon.com/blogs/developer/service-error-handling-modular-aws-sdk-js/)